subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/plugins/XCTestConventionPlugin.java

```diff
@@ -128,52 +126,36 @@ private void configureTestSuiteBuildingTasks(Project project) {
-                    testTask.onlyIf(new Spec<Task>() {
-                        @Override
-                        public boolean isSatisfiedBy(Task element) {
-                            return testTask.getTestBundleDirectory().getAsFile().get().exists();
-                        }
-                    });
-                }
-            });
-        } else if (OperatingSystem.current().isLinux()){
-            result = tasks.create("xcTest", RunTestExecutable.class, new Action<RunTestExecutable>() {
-                @Override
-                public void execute(final RunTestExecutable testTask) {
-                    final InstallExecutable installTask = (InstallExecutable) tasks.getByName("installTest");
-                    testTask.setExecutable(installTask.getRunScript());
-
-                    // TODO: Honor changes to build directory
-                    testTask.setOutputDir(project.getLayout().getBuildDirectory().dir("test-results/xctest").get().getAsFile());
-                    testTask.onlyIf(new Spec<Task>() {
-                        @Override
-                        public boolean isSatisfiedBy(Task element) {
-                            return installTask.getExecutable().exists();
-                        }
-                    });
-                }
-            });
-        }
+            Names names = Names.of(testSuite.getDevelopmentBinary().getName());
 
-        if (result != null) {
-            result.setGroup(LifecycleBasePlugin.VERIFICATION_GROUP);
-            result.setDescription("Executes XCTest suites");
+            InstallXCTestBundle installTask = tasks.create(names.getTaskName("install"), InstallXCTestBundle.class);
+            installTask.getBundle().set(((DefaultSwiftXcodeXCTestSuite)testSuite).getDevelopmentBinary().getBundleDirectory());
+            installTask.getInstallDirectory().set(project.getLayout().getBuildDirectory().dir("install/" + names.getDirName()));
+
+            testTask.getTestSuiteLocation().set(installTask.getInstallDirectory());
+            testTask.getRunScript().set(installTask.getRunScript());
+            testTask.getWorkingDirectory().set(installTask.getInstallDirectory());
+        } else if (OperatingSystem.current().isLinux()) {
+            InstallExecutable installTask = (InstallExecutable) tasks.getByName("installTest");
+            testTask.getTestSuiteLocation().set(installTask.getInstallDirectory());
+            testTask.getRunScript().set(installTask.getRunScript());
+            testTask.getWorkingDirectory().set(installTask.getInstallDirectory());
         }
-        return result;
+
+        testTask.setGroup(LifecycleBasePlugin.VERIFICATION_GROUP);
+        testTask.setDescription("Executes XCTest suites");
+        testTask.onlyIf(new Spec<Task>() {
+            @Override
+            public boolean isSatisfiedBy(Task element) {
+                return testTask.getRunScript().getAsFile().get().exists();
```

Comment:
When will this be false?

Reply:
I added the skip knowledge directly on the task type.